exchanges/orders.py

```diff
@@ -0,0 +1,146 @@
+from .assets import Asset
+
+def basic_order_check(action, asset_pair):
+    if action not in ['buy', 'sell']:
+        raise ValueError("Action must be 'buy' or 'sell'")
+
+    asset0, asset1 = asset_pair
+    if not isinstance(asset0, Asset) or not isinstance(asset1, Asset):
+        raise ValueError("Not recognized asset pair")
+
+
+class Order(object):
```

Comment:
With the current implementation, an order is just data that will be given to the `place_order` method of `Exchange`. This means that the method will have to check the order type (limit, market, ...) and respond correctly. 

As a consequence:

1. if we add an order type, all exchanges will have to be updated to handle the new case. 
2. some exchanges may not implement a certain order type and will crash at run time when given that type.

For 2, I think we can build a system like the currency one where the exchange gives you a list of orders it can handle. For 1, I don't believe there is a way to avoid some manual management, the API of exchanges will be different and we will have to write custom code for each. We just have to make sure this complexity does not bubble up to traders.